test/compiler/inline.jl

```diff
@@ -1709,3 +1709,10 @@ let src = code_typed1((Any,)) do x
     end
     @test count(iscall((src, f_union_unmatched)), src.code) == 0
 end
+
+# Test that inlining doesn't unnecesarily move things to stmt position
+@noinline f_no_inline_invoke(x::Union{Symbol, Nothing}=nothing) = Base.donotdelete(x)
+g_no_inline_invoke(x) = f_no_inline_invoke(x)
+let src = code_typed1(g_no_inline_invoke, Tuple{Any})
```

Comment:
```suggestion
let src = code_typed1(g_no_inline_invoke, Tuple{Union{Symbol, Nothing}})
```
@aviatesk I think this should make the test effective.

Reply:
Confirmed. LGTM.